s4e-backend/src/main/java/pl/cyfronet/s4e/service/SchemaService.java

```diff
@@ -49,28 +48,56 @@ public String getContentByName(String name) throws NotFoundException {
     }
 
     @Transactional
-    public Long create(Create params) throws NotFoundException, SchemaCreationException {
+    public Long create(UpdateRequest params) throws NotFoundException, SchemaTypeException {
         Schema.SchemaBuilder schemaBuilder = Schema.builder()
                 .name(params.getName())
                 .type(params.getType())
                 .content(params.getContent());
 
         if (params.getPrevious() != null) {
-            String previousName = params.getPrevious();
-            Schema previous = schemaRepository.findByName(previousName)
-                    .orElseThrow(() -> constructNFE(previousName));
+            schemaBuilder.previous(getValidatedPreviousSchema(params.getType(), params.getPrevious()));
+        }
 
-            if (previous.getType() != params.getType()) {
-                throw new SchemaCreationException("Type of created schema must be the same as the previous");
-            }
+        Schema schema = schemaRepository.save(schemaBuilder.build());
+        return schema.getId();
+    }
 
-            schemaBuilder.previous(previous);
+    @Transactional
+    public Long update(UpdateRequest params) throws NotFoundException, SchemaTypeException {
+        String name = params.getName();
+        Schema schema = schemaRepository.findByName(name)
+                .orElseThrow(() -> constructNFE(name));
+
+        if (params.getType() != null) {
+            schema.setType(params.getType());
+        }
+        if (params.getContent() != null) {
+            schema.setContent(params.getContent());
+        }
+
+        if (params.getPrevious() != null) {
+            String previousName = params.getPrevious();
+            if ("".equals(previousName)) {
```

Comment:
why not isEmpty or isBlank? its more readable, you check for null above so you can easily use one of those here.